pulsar-metadata/src/test/java/org/apache/pulsar/metadata/MetadataStoreTest.java

```diff
@@ -480,6 +481,19 @@ public void testThreadSwitchOfZkMetadataStore(boolean hasSynchronizer, boolean e
             verify.run();
             return null;
         }).join();
+        // blocking chaining call
+        store.get(prefix + "/d1").thenApply((ignore) -> {
+                    try {
+                        verify.run();
+                        return store.get(prefix + "/e1").thenApply((ignore2) -> {
+                            verify.run();
+                            return null;
+                        }).get();
```

Comment:
The problem here is a mix between sync & async calls.
eg: blocking in a callback thread. 

Using ForkJoinPool, it's only moving the problem to a different place. It will be able to avoid deadlocking on a simple 1:1 situation, though when the ForkJoinPool threads count is maxed, we will still be subject to deadlock.

No matter how we handle this, it's going to be problematic, hence we always have had to be careful in avoiding blocking calls to metadata.